src/main/java/bio/terra/cli/command/Main.java

```diff
@@ -60,18 +71,71 @@ public static void main(String... args) {
   public void run() {}
 
   /**
-   * Custom exception handler class that suppresses the stack trace when an exception is thrown.
-   * Instead, it just prints the exception message and exits the process.
+   * Custom handler class that intercepts all exceptions.
+   *
+   * <p>There are three categories of exceptions. All print a message to stderr and log the
+   * exception.
+   *
+   * <p>- User-facing = user can fix
+   *
+   * <p>- Internal = user cannot fix, exception specifically thrown by CLI code
+   *
+   * <p>- Unexpected = user cannot fix, exception not thrown by CLI code
+   *
+   * <p>The internal and unexpected cases are very similar, except that the message on the internal
+   * exception might be more readable/relevant.
    */
-  private static class PrintExceptionMessageHandler
+  private static class UserFacingExceptionHandler
       implements CommandLine.IExecutionExceptionHandler {
+
+    private static final org.slf4j.Logger logger =
+        LoggerFactory.getLogger(ch.qos.logback.classic.Logger.ROOT_LOGGER_NAME);
+
+    private CommandLine commandLine;
+
+    @Override
     public int handleExecutionException(Exception ex, CommandLine cmd, ParseResult parseResult) {
+      this.commandLine = cmd;
+
+      String errorMessage;
+      boolean printPointerToLogFile;
+      if (ex instanceof UserFacingException) {
```

Comment:
It looks like `Logger#LOG_FORMAT` is configured to prefix messages with their log level?  If so, you might consider moving the `"[ERROR] "` prefix append to `printErrorText()` time.

Reply:
Yes, you're right. Done.